src/pre_process/p_main.f90

```diff
@@ -120,11 +131,11 @@ program p_main
         end if
     end if
 
-    if (old_ic) call s_read_ic_data_files(q_cons_vf)
+    if (old_ic) call s_read_ic_data_files(q_cons_vf, pb, mv)
```

Comment:
don't like this addition

Reply:
I could make these variables global in both simulation and pre_process, which would no longer require their presence in the function arguments

Reply:
that might be a good option